sentry-core/src/main/java/io/sentry/core/SendCachedEventFireAndForgetIntegration.java

```diff
@@ -12,32 +15,59 @@
   }
 
   public interface SendFireAndForgetDirPath {
+    @Nullable
     String getDirPath();
   }
 
   public interface SendFireAndForgetFactory {
+    @Nullable
     SendFireAndForget create(IHub hub, SentryOptions options);
+
+    default boolean hasValidPath(final @Nullable String dirPath, final @NotNull ILogger logger) {
+      if (dirPath == null || dirPath.isEmpty()) {
+        logger.log(SentryLevel.INFO, "No cached dir path is defined in options.");
+        return false;
+      }
+      return true;
+    }
+
+    default @NotNull SendFireAndForget processDir(
+        final @NotNull DirectoryProcessor directoryProcessor,
+        final @NotNull String dirPath,
+        final @NotNull ILogger logger) {
+      final File dirFile = new File(dirPath);
+      return () -> {
+        logger.log(SentryLevel.DEBUG, "Started processing cached files from %s", dirPath);
+
+        directoryProcessor.processDirectory(dirFile);
+
+        logger.log(SentryLevel.DEBUG, "Finished processing cached files from %s", dirPath);
+      };
+    }
   }
 
-  public SendCachedEventFireAndForgetIntegration(SendFireAndForgetFactory factory) {
-    this.factory = factory;
+  public SendCachedEventFireAndForgetIntegration(final @NotNull SendFireAndForgetFactory factory) {
```

Comment:
Super hard to review this PR with so much unrelated changes like this

Reply:
this class needed to be changed, so it made sense to "improve" the missing finals and annotations IMO.
this class has more changes than this and adding NotNull helps with interoperability with Kotlin tests